src/textual/widgets/_collapsible.py

```diff
@@ -214,3 +224,8 @@ def compose_add_child(self, widget: Widget) -> None:
             widget: A Widget to add.
         """
         self._contents_list.append(widget)
+
+    def _watch_title(self, title: str) -> None:
+        if not self.is_mounted:
```

Comment:
Do we need this? If somebody extends this class and modifies `title` in the constructor, won't this prevent the title from updating?

AFAIK we only need this if we are using the DOM, but updating the state is not a problem.

Reply:
Here's the trackback if this is removed:

```python
$ textual run docs/examples/widgets/collapsible.py 
╭──────────────────────────────────────────────────────────── Traceback (most recent call last) ─────────────────────────────────────────────────────────────╮
│ /home/tom/Projects/textual/docs/examples/widgets/collapsible.py:33 in compose                                                                              │
│                                                                                                                                                            │
│   30 │   def compose(self) -> ComposeResult:                                                                                                               │
│   31 │   │   """Compose app with collapsible containers."""                                                                                                │
│   32 │   │   yield Footer()                                                                                                                                │
│ ❱ 33 │   │   with Collapsible(collapsed=False, title="Leto"):                                                                                              │
│   34 │   │   │   yield Label(LETO)                                                                                                                         │
│   35 │   │   yield Collapsible(Markdown(JESSICA), collapsed=False, title="Jessica")                                                                        │
│   36 │   │   with Collapsible(collapsed=True, title="Paul"):                                                                                               │
│                                                                                                                                                            │
│ ╭─────────────────────────────── locals ────────────────────────────────╮                                                                                  │
│ │ self = CollapsibleApp(title='CollapsibleApp', classes={'-dark-mode'}) │                                                                                  │
│ ╰───────────────────────────────────────────────────────────────────────╯                                                                                  │
│                                                                                                                                                            │
│ /home/tom/Projects/textual/src/textual/widgets/_collapsible.py:182 in __init__                                                                             │
│                                                                                                                                                            │
│   179 │   │   │   disabled: Whether the collapsible is disabled or not.                        ╭───────────── locals ─────────────╮                        │
│   180 │   │   """                                                                              │         children = ()            │                        │
│   181 │   │   super().__init__(name=name, id=id, classes=classes, disabled=disabled)           │          classes = None          │                        │
│ ❱ 182 │   │   self.title = title                                                               │        collapsed = False         │                        │
│   183 │   │   self._title = CollapsibleTitle(                                                  │ collapsed_symbol = '▶'           │                        │
│   184 │   │   │   label=title,                                                                 │         disabled = False         │                        │
│   185 │   │   │   collapsed_symbol=collapsed_symbol,                                           │  expanded_symbol = '▼'           │                        │
│                                                                                                │               id = None          │                        │
│                                                                                                │             name = None          │                        │
│                                                                                                │             self = Collapsible() │                        │
│                                                                                                │            title = 'Leto'        │                        │
│                                                                                                ╰──────────────────────────────────╯                        │
│                                                                                                                                                            │
│ /home/tom/Projects/textual/src/textual/widgets/_collapsible.py:229 in _watch_title                                                                         │
│                                                                                                                                                            │
│   226 │   │   self._contents_list.append(widget)                                               ╭─────── locals ────────╮                                   │
│   227 │                                                                                        │  self = Collapsible() │                                   │
│   228 │   def _watch_title(self, title: str) -> None:                                          │ title = 'Toggle'      │                                   │
│ ❱ 229 │   │   self._title.label = title                                                        ╰───────────────────────╯                                   │
│   230                                                                                                                                                      │
╰────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'Collapsible' object has no attribute '_title'
```

Reply:
I've now removed this so hopefully we're almost there.

Sorry I thought this was going to be a simple PR but I think my contributing has been more of a hindrance than help here. Thanks for your patience.

Reply:
It's not your fault. Your PR was in excellent shape when I first reviewed, I was the one that pushed you to do something else :P

Reply:
> Sorry I thought this was going to be a simple PR but I think my contributing has been more of a hindrance than help here. Thanks for your patience.

Not at all. You contributions are always welcome!